### PR TITLE
build: map in the full teamcity alternates directory

### PR DESCRIPTION
**Pull request by petermattis**
Release note: None

### CODE REVIEW

File: build/builder.sh [hunk 1]
```diff
@@ -120,10 +120,15 @@ vols="${vols} --volume=${HOME}/.yarn-cache:${container_home}/.yarn-cache${cached
...
-  vols="${vols} --volume=${alternates_path}:${alternates_path}${cached_volume_mode}"
+# alternates_file=${cockroach_toplevel}/.git/objects/info/alternates
+# if test -e "${alternates_file}"; then
+#   alternates_path=$(cat "${alternates_file}")
+#   vols="${vols} --volume=${alternates_path}:${alternates_path}${cached_volume_mode}"
+# fi
+
+teamcity_alternates="/home/agent/system/git/"
+if test -d "${teamcity_alternates}"; then
+    vols="${vols} --volume=${teamcity_alternates}:${teamcity_alternates}${cached_volume_mode}"
```
Comment (jordanlewis) [inline build/builder.sh:131]: Could we `dirname` the result of the previous version? I think that would do the trick without being as TC specific.

Comment (jordanlewis) [reply build/builder.sh:131]: So `alternates_path=$(dirname $(cat "${alternates_file}"))`

Comment (petermattis) [reply build/builder.sh:131]: Probably. I'm just looking for the sledgehammer fix here. We can follow-up with a cleaner fix if this PR actually works.

I think we could also look for `alternates` files in other `.git` directories.

**Review by jordanlewis — APPROVED** [review]
LGTM

**Comment by petermattis** [comment]
Some of the `roachtest` CI tests passed, while previously they had all failed. I'm going to get bors working. We can clean this up tomorrow.

bors r+

**Comment by petermattis** [comment]
Examples-ORMs failure. Hoping it was just a flake.

bors r+

**Pull request merged**